### PR TITLE
feat: add v2alpha3 format

### LINKED ISSUE #1026: Create v2alpha3 chains format that uses Tekton Pipeline v1 E2E natively (including provenance info( /format) + storage (/storage))
Once https://github.com/tektoncd/chains/issues/985 and https://github.com/tektoncd/chains/issues/986 are completed (via https://github.com/tektoncd/chains/pull/1016) the next step is to add a new tektoncd/chains format, `v2alpha3` that uses Tekton Pipeline V1 objects E2E with no downgrading (like is done for v2alpha2 and below to keep backwards compatibility).  This work depends on #1016 being completed

### PR DESCRIPTION
**Pull request by aaron-prindle**
Fixes #1026 

This PR adds a `v2alpha3` format which uses Tekton `v1` object(s) in it's Payload information.  For backwards compatibility, v2alpha2 and below versions use Tekton `v1beta1` object(s) in their Payload information so this is first/only format currently that uses Tekton `v1` objects E2E (doesn't convert to v1beta1 in `format/*`)

This PR is the second part of the in the v1beta1 -> v1 tektoncd/chains work split that was identified here:
https://github.com/tektoncd/chains/pull/1016#pullrequestreview-1789377427

Once this is in, the workstream there should be complete

### CODE REVIEW

File: pkg/chains/verifier.go [hunk 1]
```diff
@@ -52,12 +52,25 @@ func (tv *TaskRunVerifier) VerifyTaskRun(ctx context.Context, tr *v1.TaskRun) er
...
-	}
 
-	trObj := objects.NewTaskRunObjectV1Beta1(trV1Beta1)
+	// if cfg.Artifacts.TaskRuns.Format == "v2alpha3" && (cfg.Artifacts.PipelineRuns.Format != "" || cfg.Artifacts.PipelineRuns.Format != "v2alpha3") {
+	// 	return errors.New("When using v2alpha3, artifacts.taskrun.format and artifacts.pipelinerun.format must be identical if both are set")
+	// }
+
+	// if cfg.Artifacts.PipelineRuns.Format == "v2alpha3" && (cfg.Artifacts.TaskRuns.Format != "" || cfg.Artifacts.TaskRuns.Format != "v2alpha3") {
+	// 	return errors.New("When using v2alpha3, artifacts.taskrun.format and artifacts.pipelinerun.format must be identical if both are set")
+	// }
```
Comment (chitrangpatel) [inline pkg/chains/verifier.go:62]: zap the commented lines.

**Review by chitrangpatel — APPROVED** [review]

File: pkg/chains/verifier.go (hunk 1 above)
Comment (aaron-prindle) [reply pkg/chains/verifier.go]: Done, thanks for catching this!

**Comment by aaron-prindle** [comment]
Thanks @wlynch for the reviews on https://github.com/tektoncd/chains/pull/1016.  This PR is the second part of the in the v1beta1 -> v1 tektoncd/chains work split that was identified here:
https://github.com/tektoncd/chains/pull/1016#pullrequestreview-1789377427 

Once this is in, the workstream there should be complete

**Review by vdemeester — APPROVED** [review]
❤️

**Review by wlynch — COMMENTED** [review]
Overall looks good!

File: pkg/chains/formats/slsa/v2alpha3/internal/external_parameters/external_parameters.go
[the remark this answers is not in the document]
Comment (aaron-prindle) [reply pkg/chains/formats/slsa/v2alpha3/internal/external_parameters/external_parameters.go:1]: The files are not duplicate, one uses `v1` Tekton objects, the other uses `v1beta1` objects.  

There was a single dupe method though, `buildConfigSource` that I have now refactored by moving it into `.../formats/slsa/v2/internal/externalparameters/[external_parameters.go|external_parameters_test.go]` and referring that method across `v2alpha2` and `v2alpha3`.

File: docs/config.md [hunk 2]
```diff
@@ -21,20 +21,20 @@ Supported keys include:
 
 | Key                         | Description                                                                                                                                                                                      | Supported Values                           | Default   |
 | :-------------------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :----------------------------------------- | :-------- |
-| `artifacts.taskrun.format`  | The format to store `TaskRun` payloads in.                                                                                                                                                       | `in-toto`, `slsa/v1`, `slsa/v2alpha2`      | `in-toto` |
```
[the remark this answers is not in the document]
Comment (aaron-prindle) [reply docs/config.md:24]: I have updated the documentation here to have both `slsa/v2alpha2` and `slsa/v2alpha3` with `NOTE`s explaining the difference in Tekton API Object usage (v1beta1 vs v1) as well as the recommended version for new chains users (v2alpha3)

**Comment by aaron-prindle** [comment]
I believe all review feedback has been addressed now, lmk if there is any additional feedback or items to address.  Thanks!

File: pkg/chains/formats/slsa/v2alpha3/internal/external_parameters/external_parameters.go
[the remark this answers is not in the document]
Comment (aaron-prindle) [reply pkg/chains/formats/slsa/v2alpha3/internal/external_parameters/external_parameters.go:1]: Additionally de-duped internal_parameters/, resolved_dependecies/, and build_type (all dupe locations)

**Review by wlynch — COMMENTED** [review]
I think there's probably more refactoring we can do to leverage TektonObject to reduce duplication across v2alpha2 and v2alpha3, but I won't let this block the PR. LGTM!

/lgtm

**Pull request merged**